cancat/uds/test.py

```diff
@@ -0,0 +1,141 @@
+import time
+import struct
+
+from cancat.uds import NegativeResponseException
+from cancat.uds.ecu import ScanClass
+
+class CanInterface(object):
+    """
+    A fake CanInterface class for test purposes.
+
+    All actual functions are just stubbed out, this is not a superclass of the
+    real cancat.CanInterface class.
+    """
+    def __init__(self, *args, **kwargs):
+        # required to make canmap think the bus is working
+        self.msg_count = 1
+
+    def setCanBaud(self, *args, **kwargs):
+        pass
+
+    def placeCanBookmark(self, *args, **kwargs):
+        pass
+
+    def saveSessionToFile(self, *args, **kwargs):
+        pass
+
+    def getCanMsgCount(self):
+        count = self.msg_count
+        self.msg_count += 1
+        return count
+
+
+class TestUDS(ScanClass):
```

Comment:
this looks like a great module for use in unit tests.  
could you please make a unit test for UDS and scanning?

create a directory cancat/tests/
and a module cancat/tests/test_uds.py

with somethink like:

import unittest
class UDStest(unittest):
....
    def test_blah(self):
        self.assertEqual(self.blah, "magic comparison")
    def test_foo(self):
...

Reply:
I will add some unit tests for canmap